custom_components/rpi_gpio/binary_sensor.py

```diff
@@ -105,10 +107,20 @@ def setup_platform(
 class RPiGPIOBinarySensor(BinarySensorEntity):
     """Represent a binary sensor that uses Raspberry Pi GPIO."""
 
-    async def async_read_gpio(self):
-        """Read state from GPIO."""
-        await asyncio.sleep(float(self._bouncetime) / 1000)
-        self._state = await self.hass.async_add_executor_job(read_input, self._port)
+    @callback
+    async def async_added_to_hass(self):
+        # Arm the timer
+        timer_cancel = async_track_time_interval(
+            self.hass,
+            self.update_gpio,   # launch every second
+            interval=timedelta(seconds=1),
+        )
```

Comment:
Seems like a bit of a shame to poll every second, I wonder if libgpiod2 wouldn't allow us to get a smarter notification somehow - but the old implementation also polled, so this does not seem worse :)